source/module_cell/module_symmetry/symmetry_basic.h

```diff
@@ -37,15 +37,17 @@ class Symmetry_Basic
 			const ModuleBase::Vector3<double> &bb2,
 			const ModuleBase::Vector3<double> &bb3
 			);
-	void matrigen(ModuleBase::Matrix3 *symgen, const int ngen, ModuleBase::Matrix3* symop, int &nop);
-	void setgroup(ModuleBase::Matrix3 *symop, int &nop, const int &ibrav);
-	void rotate( ModuleBase::Matrix3 &gmatrix, ModuleBase::Vector3<double> &gtrans, int i, int j, int k, const int, const int, const int, int&, int&, int&);
-
+	void matrigen(ModuleBase::Matrix3 *symgen, const int ngen, ModuleBase::Matrix3* symop, int &nop) const;
+	void setgroup(ModuleBase::Matrix3 *symop, int &nop, const int &ibrav) const;
+	void rotate(
+			ModuleBase::Matrix3 &gmatrix, ModuleBase::Vector3<double> &gtrans, 
+			int i, int j, int k, const int, const int, const int, int&, int&, int&);
 	protected:
 
-	std::string get_brav_name(const int ibrav);
+	std::string get_brav_name(const int ibrav) const;
 	void pointgroup(const int &nrot,int &pgnumber,std::string &pgname, const ModuleBase::Matrix3* gmatrix, std::ofstream &ofs_running);
 	void atom_ordering(double *posi, const int natom, int *subindex);
+	void atom_ordering_new(double *posi, const int natom, int *subindex);
```

Comment:
Could you please add unit test for newly added functions?

Reply:
Done in the latest commit.